src/proposal.mligo

```diff
@@ -59,9 +59,8 @@ let ensure_proposal_is_unique (propose_params, store : propose_params * storage)
 
 [@inline]
 let check_is_proposal_valid (config, propose_params, store : config * propose_params * storage): storage =
```

Comment:
There is a single use of `check_is_proposal_valid` now, in `propose`:
```ocaml
  let store = check_is_proposal_valid (config, param, store) in
```
I would remove this function's declaration and instead replace its use with:
```ocaml
  let _ : unit = config.proposal_check (propose_params, store.extra)
```